Vnext Beta

KenAJoh opened the pull request:
## Pakker
- Nye pakker blir publisert under scope `@navikt` med prefix `ds-` for pakkenavn.
- `@navikt/ds-react`, Alle react komponenter skrevet om til oppdatert syntax, bruker ref-forwarding, engelsk api, endret api
- `@navikt/ds-css`, Alle komponenter skrevet om fra .less til .css, bruker CSS.
- `@navikt/ds-icons`, Ikoner laget av designsystemet(Wei) publisert som react-komponenter og svg.
- `@navikt/ds-tokens`, Design tokens publisert gjennom css-variabler, less, js og sass.

## Versjonering
- Bare `@navikt/ds-icons` vi bli publisert først med v0.1.0

## Dokumentasjon
- Design.nav.no/beta 
### Ny ikonside med søk
https://design.nav.no/resources/icons/

KenAJoh committed:
base setup

KenAJoh committed:
ds-tokens

KenAJoh committed:
autofocus on popover

KenAJoh committed:
fixed styles

KenAJoh committed:
satt opp api

KenAJoh committed:
oppdateres riktig når den først åpnes nå

andnorda committed:
popover doesn't focus

KenAJoh committed:
fixed focus handling

andnorda committed:
design tokens proposal

KenAJoh committed:
lint fix

KenAJoh committed:
Merge branch 'vnext-popover' of https://github.com/navikt/nav-frontend-moduler into vnext-popover

KenAJoh committed:
Merge remote-tracking branch 'origin/vnext-popover-design-tokens' into vnext-popover

andnorda committed:
design tokens proposal (#875)

KenAJoh committed:
css fix, token watch

KenAJoh committed:
refactored popover component

KenAJoh committed:
readded fokusfarmge¨

KenAJoh committed:
oppdateringer

KenAJoh committed:
beta version

KenAJoh committed:
fix merge conflict

KenAJoh committed:
lint fix

KenAJoh committed:
deps popover

KenAJoh committed:
popover-style deps

KenAJoh committed:
tokens deps

KenAJoh committed:
la til metadata i package.json

KenAJoh committed:
story fix

KenAJoh committed:
css prefixing

KenAJoh committed:
lint fix

KenAJoh committed:
readded stories2

KenAJoh committed:
la til sourcemap

KenAJoh committed:
style -> styles

KenAJoh committed:
Merge branch 'master' into vnext-popover

KenAJoh committed:
fixed beta versjoner

KenAJoh committed:
readded nodemon

KenAJoh committed:
fix index

andnorda committed:
update popover stories import

andnorda committed:
popover rename

andnorda committed:
delete old readmes

andnorda committed:
Vnext button (#878)

* wip

* as=a and ref forwarding

* handle link with className

* refactor css

* css variables cleanup

* tokens cleanup

* rename button -> react-button

* center-align text

* refactor story

* oopsie

* ender story button-links

Co-authored-by: KenAJoh <ken.aleksander.johansen@nav.no>

KenAJoh committed:
popover -> react-popover

andnorda committed:
make a chindren explicit

KenAJoh committed:
Merge branch 'master' into vnext

KenAJoh committed:
new yarn lock

andnorda committed:
Merge branch 'master' of github.com:navikt/nav-frontend-moduler

andnorda committed:
Merge branch 'master' into vnext

andnorda committed:
Vnext alert + icons (#886)

* wip

* as=a and ref forwarding

* handle link with className

* refactor css

* css variables cleanup

* tokens cleanup

* rename button -> react-button

* center-align text

* refactor story

* oopsie

* add icons

* icon stories

* add alert

* Filled icons

* adjust colors

* refactored color tokens

* border and background tokens

* restructure tokens

* fix typo

andnorda committed:
i prefer the word elevation over lift

andnorda committed:
Compact proposal (#887)

* First, simplistic compact proposal

* second, more refined proposal

andnorda committed:
more BEM like popover hidden state class name

andnorda committed:
Vnext token improvements (#888)

* add text color tokens

* use text inverse color in button

KenAJoh committed:
Vnext scaffolding script (#891)

* fixed scaffolding

* added prettierignore

* updated config

* updated css

* oppdatert scaffolding

* updated scaffolding

* updated scaffolding2

* updated scaffolding2

* scaffolding react fullført

* react og css scaffolding fullført

* cleanup

* utbedret scaffolding script

* fikset style import i template

KenAJoh committed:
Vnext-tag (Etikett) (#894)

* fixed scaffolding

* added prettierignore

* updated config

* updated css

* oppdatert scaffolding

* updated scaffolding

* updated scaffolding2

* updated scaffolding2

* scaffolding react fullført

* react og css scaffolding fullført

* cleanup

* utbedret scaffolding script

* wcaffold setup

* fjernet uferdig scafolding

* fikset scaffolding

* la til css dist for nav-frontend-etikett, implemented Tag component

* ordnet stiler, fikset scaffolding imports

* fikset package.json scaffolding

* fullført tag component

* la til dokumentasjon for props

* oppdatert tag styling og tokens

KenAJoh committed:
Vnext typography (#904)

* first attempt at a typography setup

* add baseline-styles package

* tokens

* fjernet lineheight text

* la til fonter i baseline

* fjernet list-style

* oppdatert tag-component

* fikset tag story

* oppdatert line-height tokens

* oppdatert lin-height

* justering alert css

* oppdatert alert

* oppdatert button font

Co-authored-by: Andreas Nordahl <andnorda@gmail.com>

KenAJoh committed:
Vnext accordion, chevron (#889)

* oppsett

* start accordion css

* Implementert chevron kompnent

* fullført accordion styles

* border fokus fiks

* chevron i tag -> span

* justert accordion API

* Oppdaterte accordion til å bruke icons

* oppdatert imports

* småjustering css, fjernet border prop

* fjernet css no-border test

* fjernet unødvendig test, la til åpen test

* styling renamet, fjernet duplicate css

* fjernet buttonId props

* fjernet ubrukt onRest, la til onRest direkte som prop

* fikset imports, story

* add css styleline config (#898)

* fjernet onRest prop

* fixed merge issues

* la til stylelint regel

* css endringer

* forenklet kode

* oppdatert accordion

* fikset chevron for accordion

Co-authored-by: Andreas Nordahl <andnorda@gmail.com>

KenAJoh committed:
Vnext link (Lenke) (#900)

* scaffolded link

* implemented link react-componetn

* Make href not optional, added story

* fullførte story

* implemented styles

* oppdatert story

* fikset href link

* link-styles bruker nå css-variabler

* Fjernet unødvendig css

* endriner css

* fikset ikon stilen

KenAJoh committed:
Vnext modal (#899)

* scaffolded modal

* added modal styles

* added props to modal

* merged refs modal

* modal er nå funksjonell

* justeringer api

* fullført modal-component

* modal-styles bruker variabler nå

* added todo

* La til z-index token

* endringer for tokens

* Api endringer

* Fikset stories

* fjernet ubrukt prop

* Fikset tokens

* oppdatert yarn lock

KenAJoh committed:
Vnext typo endringer og litt småfiks + qol (#905)

* Flyttet nye komponenter i sorterte dirs

* oppdatert accordion

* justerte alert-styles

* oppdatert button-styles

* oppdatert variabler i accordion og modal

* oppdatert tag-styles

* oppdatert typo-styles

* oppdatert typo-styles

* Vnext react pakke (draft) (#907)

* satt opp react pakke

* react pakke cleanup

* ferdigsatt react-pakke

KenAJoh committed:
Vnext pakkeoppsett  (#909)

* Flyttet nye komponenter i sorterte dirs

* oppdatert accordion

* justerte alert-styles

* oppdatert button-styles

* oppdatert variabler i accordion og modal

* oppdatert tag-styles

* oppdatert typo-styles

* oppdatert typo-styles

* Vnext react pakke (draft) (#907)

* satt opp react pakke

* react pakke cleanup

* ferdigsatt react-pakke

* oppdatert style-pakke, fikset react til å bruke denne pakken

* ferdig css pakke

* pakker nå satt opp for release

* fjernet react- prefix

* fikset modal import

* la til forenklet package.json for komponenter

* Fikset merge conflict

andnorda committed:
oppsett for beta doc (#908)

andnorda committed:
hide beta from menu

KenAJoh committed:
fikset pakkenavn (#911)

KenAJoh committed:
Litt småfiks for dependencies (#912)

* fikset pakkenavn

* justering dependencies

* fikset deps css

KenAJoh committed:
Merge branch 'master' into vnext

KenAJoh committed:
lint fix

andnorda commented on @navikt/ds-css/modal/index.css:
```diff
@@ -0,0 +1,56 @@
+
+  background-color: var(--navds-color-white);
+  display: block;
+  padding: 1rem 1rem 1rem 1rem;
```
```suggestion
  padding: 1rem;
```

andnorda commented on @navikt/ds-css/modal/index.css:
```diff
@@ -0,0 +1,56 @@
+  padding: 1rem 1rem 1rem 1rem;
+  border-radius: var(--modal-border-radius);
+  position: relative;
+  flex-grow: 0;
```
Usikker på om vi faktisk trenger denne.

KenAJoh replied:
👍  ser ut som den defaulter til 0 uansett

andnorda commented on @navikt/ds-css/modal/index.css:
```diff
@@ -0,0 +1,56 @@
+}
+.navds-modal:focus,
+.navds-modal--focus {
+  box-shadow: var(--navds-shadow-focus);
```
Jeg trodde ikke selve modalen var et interaktivt element. Skal den ha ha focus-stil? Ref. det vi snakket om for popover.

KenAJoh replied:
Slik modalen er nå idag så vil man sette fokus på hele modal-elementet når man klikker på det. Det er også default at man får fokus på hele modal når den blir åpnet. Tenker dette er noe som er laget for å tilfredstille skjermlesere. 

Når det kommer til hvordan vi setter fokus når `shouldCloseOnOverlayClick` er false er jeg enda ikke sikker på. Vi kan velge å ikke gjøre noe som helst med fokuset her, tror kanskje det er den beste løsningen for nå.

andnorda commented on @navikt/ds-css/modal/index.css:
```diff
@@ -0,0 +1,56 @@
+  right: 1rem;
+}
+.navds-modal__button svg {
+  /* TODO: Set to 1rem when close icon is fixed to 16x16px */
```
Skal vi fikse dette før vi merger?

KenAJoh replied:
Burde helst være fikset ja, men har en mistanke om at ikonet ikke vil se bra ut hvis man bare gjør strekene lengre for å nå 16x16 px. Får høre med Wei

andnorda commented on @navikt/ds-icons/svg/Bookmark-1.svg:
```diff
@@ -0,0 +1,5 @@
+<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Skal vi fikse ordentlige navn på ikon-varianter før vi merger?

andnorda replied:
Jeg er usikker på om det er best å gjøre det når vi eksporterer, eller når vi bygger pakken.

andnorda replied:
Eller kanskje det finnes helt andre måter å løse det på.

KenAJoh replied:
Ja vi burde kanskje finne en løsning på dette først.. Får tenke over eventuelle løsninger, men siden SVG ene også exponeres gjennom pakken må de har riktige navn å.

andnorda replied:
Godt poeng

andnorda commented on @navikt/ds-css/accordion/index.css:
```diff
@@ -0,0 +1,75 @@
+}
+.navds-accordion__button .navds-accordion__title {
+  font-size: var(--navds-font-size-large);
+  line-height: var(--navds-font-line-height-xs);
```
Jeg foreslår `line-height: 1` (uten enhet) her.

andnorda commented on @navikt/ds-css/accordion/index.css:
```diff
@@ -0,0 +1,75 @@
+  font-size: var(--navds-font-size-large);
+  line-height: var(--navds-font-line-height-xs);
+  font-weight: var(--navds-font-weight-bold);
+  color: var(--navds-color-text-primary);
```
Trenger vi sette denne? Burde vi i så fall sette den for andre komponenter også?

KenAJoh replied:
Trenger nok ikke å være med nei

andnorda commented on @navikt/ds-css/baseline/index.css:
```diff
@@ -0,0 +1,16 @@
+  font-size: 1.125rem;
+}
+
+a {
```
Burde denne egentlig være med? Jeg tenker jo vi helst vil at folk bruker link-komponenten, men det gir kanskje mening å ha default styles som gir mening som fallback? Burde vi i så fall lage flere grunnstiler som gir mening? F.eks. h1-h6, input, button, etc.

KenAJoh replied:
Hmm, vi burde kanskje gå gjennom nav-frontend-core å se gjennom hva vi ønsker å ta med oss før vi går live. Er mye diverse løsninger som fikser småting, samt styles for print etc

andnorda replied:
Ja, kan ta en runde gjennom de gamle filene på det. Men jeg stemmer for at vi tar med minst mulig. Jeg tror det er lurt å fjerne denne `a`-stilen også, og heller legge den til igjen om det mot formodning skulle dukke opp et ønske om det.

andnorda replied:
Jeg gikk gjennom core.less nå, og jeg fant egentlig ingenting jeg tror er kritisk, annet enn muligens `.sr-only` og eventuelt en print.css. Hva tenker du om de?

KenAJoh replied:
Tror `.sr-only` og print er greit å ta med videre da de løser reelle problemer for eks skjemlesere og utskrift av infosider/søknaer

andnorda replied:
Jeg ser at alle lenkene i `print.less` er døde, men jeg fant den samme fila som de kopierte fra i 2017 fortsatt er nesten lik (https://github.com/h5bp/html5-boilerplate/blob/master/dist/css/style.css). Skal vi bare kopiere den nyeste versjonen, eller skal vi prøve å spikke noe helt nytt?

KenAJoh replied:
Kan kanskje bare kopiere den, da å lage en ny vil ta litt tid regner jeg med.

andnorda replied:
https://github.com/navikt/nav-frontend-moduler/pull/920

andnorda reviewed (dismissed):
Vi kan gjerne bare merge med en gang for min del, og så fikse ting senere. Opp til deg.

KenAJoh committed:
Update @navikt/ds-css/modal/index.css

Co-authored-by: Andreas Nordahl <andnorda@gmail.com>

andnorda committed:
add simple typo doc of proposed structure (#914)

* add simple typo doc of proposed structure

* only scroll content

* remove unused import

andnorda committed:
add print.css and .sr-only styling (#920)

KenAJoh committed:
Justering på stylings tatt opp i vnext-beta (#921)

* fjernet flex-grow modal

* forbedret accordion transition

* line-height accordion title

* color accordion title2

* endret ikonstørrelse modal

KenAJoh committed:
Vnext master sync (#942)

* Endre Chevron til <span> fra <i> (#916)

* endret chevron til span fra i

* endret chevron til v1

* endret changelog

* Lerna versionbump [ci skip]

 - nav-frontend-chevron-style@1.0.1
 - nav-frontend-chevron@1.0.22
 - nav-frontend-ikonknapper@1.0.26
 - nav-frontend-lesmerpanel@0.0.57

* Lerna versionbump [ci skip]

* Oppdatert kontraster kode-eksempel (#918)

* Oppdatert pakker til å bruk v1 nav-frontend-chevron-styles, changelog (#919)

* endret chevron til span fra i

* endret chevron til v1

* endret changelog

* oppdatert dependencies for bruk av stilpakker

* oppdatert changelog

* Lerna versionbump [ci skip]

 - nav-frontend-chevron@1.0.23
 - nav-frontend-ekspanderbartpanel-style@1.0.32
 - nav-frontend-ekspanderbartpanel@3.0.43
 - nav-frontend-ikonknapper@1.0.27
 - nav-frontend-lenkepanel-style@0.3.43
 - nav-frontend-lenkepanel@1.0.62
 - nav-frontend-lesmerpanel@0.0.58

* Lerna versionbump [ci skip]

* Update README.md

* Lerna versionbump [ci skip]

 - nav-frontend-chevron@1.0.24
 - nav-frontend-ekspanderbartpanel-style@1.0.33
 - nav-frontend-ekspanderbartpanel@3.0.44
 - nav-frontend-ikonknapper@1.0.28
 - nav-frontend-lenkepanel-style@0.3.44
 - nav-frontend-lenkepanel@1.0.63
 - nav-frontend-lesmerpanel@0.0.59

* Lerna versionbump [ci skip]

* forsøk på fiksing av fargeside (#923)

* Lerna versionbump [ci skip]

 - nav-frontend-ekspanderbartpanel@3.0.45
 - nav-frontend-ikonknapper@1.0.29
 - nav-frontend-lenkepanel@1.0.64
 - nav-frontend-lesmerpanel@0.0.60

* Fikset lasting av farger (#924)

* forsøk på fiksing av fargeside

* fikset farge-lasting

* Oppdatert dokumentasjon for typo-komponent (#926)

* forsøk på fiksing av fargeside

* fikset farge-lasting

* oppdatert docs typografi

* Småjustering UU for nettside (#915)

* svg uten fokus

* småfiks uu

* Sidetittel i header endres til `h2` for artikkelsider (#928)

* h2 header på artikkelsider

* fjernet prop

* Fikset textarea less mixin (#930)

* Lerna versionbump [ci skip]

 - nav-frontend-skjema-style@2.0.24
 - nav-frontend-skjema@3.0.39

* Lerna versionbump [ci skip]

* Fikset Less math "mixins" for Less v4 (#937)

* fixed chevron floor

* prettier fix

* Lerna versionbump [ci skip]

 - nav-frontend-chevron-style@1.0.2
 - nav-frontend-skjema@3.0.40

* Oppdatert CRA og webpack examples (#931) [ci skip]

* removed package-lock to avoid security issues

* oppdatert CRA

* fjernet css

* oppdatert webpack og readmes

* ignorerte examples eslint

* fjernet sketch-linker fra setup

* Oppdatert deps for snyk (#938)

* fixed chevron floor

* prettier fix

* oppdaterte husky setup

* fikset undertextBold typo

* fjernet dep

* UU-fix for Alertstripe ikoner (#940)

* fikset UU for alertstripe SVG

* oppdatert changelog

* Lerna versionbump [ci skip]

 - nav-frontend-alertstriper@3.0.39

* la til location prop i header

* Link to new figma icons doc (#941)

* Link to new figma icons doc

* Add back link to old icon document

* Clarify streamline license text

Co-authored-by: Andreas Nordahl <andnorda@gmail.com>

KenAJoh committed:
Automatisert nedlastning av ikoner fra figma (#927)

* Endre Chevron til <span> fra <i> (#916)

* endret chevron til span fra i

* endret chevron til v1

* endret changelog

* Lerna versionbump [ci skip]

 - nav-frontend-chevron-style@1.0.1
 - nav-frontend-chevron@1.0.22
 - nav-frontend-ikonknapper@1.0.26
 - nav-frontend-lesmerpanel@0.0.57

* Lerna versionbump [ci skip]

* Oppdatert kontraster kode-eksempel (#918)

* Oppdatert pakker til å bruk v1 nav-frontend-chevron-styles, changelog (#919)

* endret chevron til span fra i

* endret chevron til v1

* endret changelog

* oppdatert dependencies for bruk av stilpakker

* oppdatert changelog

* Lerna versionbump [ci skip]

 - nav-frontend-chevron@1.0.23
 - nav-frontend-ekspanderbartpanel-style@1.0.32
 - nav-frontend-ekspanderbartpanel@3.0.43
 - nav-frontend-ikonknapper@1.0.27
 - nav-frontend-lenkepanel-style@0.3.43
 - nav-frontend-lenkepanel@1.0.62
 - nav-frontend-lesmerpanel@0.0.58

* Lerna versionbump [ci skip]

* Update README.md

* Lerna versionbump [ci skip]

 - nav-frontend-chevron@1.0.24
 - nav-frontend-ekspanderbartpanel-style@1.0.33
 - nav-frontend-ekspanderbartpanel@3.0.44
 - nav-frontend-ikonknapper@1.0.28
 - nav-frontend-lenkepanel-style@0.3.44
 - nav-frontend-lenkepanel@1.0.63
 - nav-frontend-lesmerpanel@0.0.59

* Lerna versionbump [ci skip]

* forsøk på fiksing av fargeside (#923)

* Lerna versionbump [ci skip]

 - nav-frontend-ekspanderbartpanel@3.0.45
 - nav-frontend-ikonknapper@1.0.29
 - nav-frontend-lenkepanel@1.0.64
 - nav-frontend-lesmerpanel@0.0.60

* Fikset lasting av farger (#924)

* forsøk på fiksing av fargeside

* fikset farge-lasting

* Oppdatert dokumentasjon for typo-komponent (#926)

* forsøk på fiksing av fargeside

* fikset farge-lasting

* oppdatert docs typografi

* Henter nedlastnings link til alle ikoner i frame

* downloads icons from cloud

* checks for folder first now

* before refactor

* before refactor

* chunked icon download

* implementerte nedlasting fra figma

* pr fikses

* mkdir om not exists

KenAJoh committed:
Generering av metadata for ikoner (#944)

* Endre Chevron til <span> fra <i> (#916)

* endret chevron til span fra i

* endret chevron til v1

* endret changelog

* Lerna versionbump [ci skip]

 - nav-frontend-chevron-style@1.0.1
 - nav-frontend-chevron@1.0.22
 - nav-frontend-ikonknapper@1.0.26
 - nav-frontend-lesmerpanel@0.0.57

* Lerna versionbump [ci skip]

* Oppdatert kontraster kode-eksempel (#918)

* Oppdatert pakker til å bruk v1 nav-frontend-chevron-styles, changelog (#919)

* endret chevron til span fra i

* endret chevron til v1

* endret changelog

* oppdatert dependencies for bruk av stilpakker

* oppdatert changelog

* Lerna versionbump [ci skip]

 - nav-frontend-chevron@1.0.23
 - nav-frontend-ekspanderbartpanel-style@1.0.32
 - nav-frontend-ekspanderbartpanel@3.0.43
 - nav-frontend-ikonknapper@1.0.27
 - nav-frontend-lenkepanel-style@0.3.43
 - nav-frontend-lenkepanel@1.0.62
 - nav-frontend-lesmerpanel@0.0.58

* Lerna versionbump [ci skip]

* Update README.md

* Lerna versionbump [ci skip]

 - nav-frontend-chevron@1.0.24
 - nav-frontend-ekspanderbartpanel-style@1.0.33
 - nav-frontend-ekspanderbartpanel@3.0.44
 - nav-frontend-ikonknapper@1.0.28
 - nav-frontend-lenkepanel-style@0.3.44
 - nav-frontend-lenkepanel@1.0.63
 - nav-frontend-lesmerpanel@0.0.59

* Lerna versionbump [ci skip]

* forsøk på fiksing av fargeside (#923)

* Lerna versionbump [ci skip]

 - nav-frontend-ekspanderbartpanel@3.0.45
 - nav-frontend-ikonknapper@1.0.29
 - nav-frontend-lenkepanel@1.0.64
 - nav-frontend-lesmerpanel@0.0.60

* Fikset lasting av farger (#924)

* forsøk på fiksing av fargeside

* fikset farge-lasting

* Oppdatert dokumentasjon for typo-komponent (#926)

* forsøk på fiksing av fargeside

* fikset farge-lasting

* oppdatert docs typografi

* Henter nedlastnings link til alle ikoner i frame

* downloads icons from cloud

* checks for folder first now

* before refactor

* before refactor

* chunked icon download

* implementerte nedlasting fra figma

* pr fikses

* mkdir om not exists

* generer metadata ved nedlastning av ikoner

* La til metadata fil

* merge fix

KenAJoh committed:
Merge branch 'master' into vnext

KenAJoh committed:
yarn lock update

KenAJoh committed:
satte scopede pakker til private (utenom icons)

KenAJoh committed:
fjernet cypress

KenAJoh committed:
Vnext scss med variabler(tokens) (#933)

* added scss tokens

* accordion scss

* accordion with mixin example

* scss med bare variabler fra tokens

* fjernet mixin

* småendringer node-sass

* småendringer node-sass2

* yarn lock update

KenAJoh committed:
Merge branch 'master' into vnext

KenAJoh committed:
ds-icons satt til private

KenAJoh committed:
fjernet scaffolding

KenAJoh committed:
fikset link story

KenAJoh committed:
fjernet duplicate stories

KenAJoh committed:
fjernet storybook build

KenAJoh committed:
fjernet react import i root

KenAJoh committed:
småjustering versjonering før merge

KenAJoh committed:
delay på popover story
